File: server/src/main/java/com/willkamp/vial/api/ResponseBuilder.java
package com.willkamp.vial.api;

import io.netty.handler.codec.http.HttpResponseStatus;
import io.netty.util.AsciiString;

public interface ResponseBuilder {

  ResponseBuilder setStatus(HttpResponseStatus status);

  ResponseBuilder setBodyJson(Object serializeToJson);

  ResponseBuilder setBodyHtml(String html);

  ResponseBuilder setBodyData(String contentType, byte[] data);

  ResponseBuilder setBodyText(String text);

  ResponseBuilder addHeader(CharSequence key, CharSequence value);

  ResponseBuilder addHeader(AsciiString key, AsciiString value);
}

File: server/src/main/java/com/willkamp/vial/api/ResponseBuilder.kt
package com.willkamp.vial.api

import io.netty.handler.codec.http.HttpResponseStatus
import io.netty.util.AsciiString

interface ResponseBuilder {

    fun setStatus(status: HttpResponseStatus): ResponseBuilder

    fun setBodyJson(serializeToJson: Any): ResponseBuilder

    fun setBodyHtml(html: String): ResponseBuilder

    fun setBodyData(contentType: String, data: ByteArray): ResponseBuilder

    fun setBodyText(text: String): ResponseBuilder

    fun addHeader(key: CharSequence, value: CharSequence): ResponseBuilder

    fun addHeader(key: AsciiString, value: AsciiString): ResponseBuilder
}
